### ISSUE TITLE
[BUG] Update Tutorial

### ISSUE THREAD

**Issue by BradGreig** [21cmfast/21cmFAST#59]
**Describe the bug:**
A few people have contacted me with issues regarding the tutorials. It seems that the tutorials no longer work as described in the documentation.

**To Reproduce:**
Explicitly follow the tutorials and it'll fail at the spin_temperature example.

**Comment by BradGreig** [comment]
In the C code, it enters the this_spin_temp->first_box condition for redshift < z_HEAT_MAX. It appears like it is attempting to fill the previous_spin_temp boxes which haven't been declared at any point.

**Comment by steven-murray** [comment]
Hi @BradGreig, I've just run the tutorial, and it ran through completely fine for me :-/ 

Here's a potential MWE:
```python
import py21cmfast as p21c

perturbed_field = p21c.perturb_field(
    redshift = 8.0,
    user_params = {"HII_DIM": 50, "BOX_LEN": 100},
    cosmo_params = p21c.CosmoParams(SIGMA_8=0.8),
    direc='_cache'  # use a unique cache just for this MWE
)

spin_temp = p21c.spin_temperature(
    perturbed_field = perturbed_field,
    z_step_factor=1.05,
    direc="_cache"
)
```

Can you confirm this doesn't work for you?

**Comment by steven-murray** [comment]
Though it seems to work fine for me, the offending lines must be the if statement from   https://github.com/21cmfast/21cmFAST/blob/abdcbddae8ffd603256ef75a45809a182d847524/src/py21cmfast/src/SpinTemperatureBox.c#L339.

It asks whether this is the first box, then inside that loop checks stuff in the `previous_spin_temp`. But if this is the first box, the previous spin temp must just be a dummy, and not have anything useful in it. 

In my understanding, the "first box" is the box with the highest redshift _still below_ `Z_HEAT_MAX`. When calculating this box, the "previous box" is created as a dummy (at redshift zero) and it is expected that it should not be used in the C at all.

**Comment by BradGreig** [comment]
@steven-murray I'll get back to you on this one. I'll need to ask the people who have had issues in the past.

**Comment by BradGreig** [comment]
@steven-murray your MWE works for me. However, this is not the order for how the tutorial is to be run. This fails for me:
```
initial_conditions = p21c.initial_conditions(                                                                                                                              
    user_params = p21c.UserParams(HII_DIM=100,BOX_LEN=100,DIM=200),
    cosmo_params = p21c.CosmoParams(SIGMA_8=0.8),
    random_seed = 1)

perturbed_field = p21c.perturb_field(
    redshift = 8.0,
    init_boxes = initial_conditions
)

spin_temp = p21c.spin_temperature(
    perturbed_field = perturbed_field,
    z_step_factor=1.05
)
```

Yes, that is exactly where the issue is. However, I'm not sure that this is the desired behaviour. If redshift < Z_HEAT_MAX then the boundary condition for the evolution should be set to the values at redshift = Z_HEAT_MAX. Because at the current redshift you must evolve from the previous redshift (which would be at redshift = Z_HEAT_MAX). So the `previous_spin_temp` should always be accessible memory space.

**Comment by steven-murray** [comment]
Hmmm, this is very strange. Why does it fail if you run the `initial_conditions` explicitly, but not if you run it implicitly within the `perturbed_field`? In both cases, the call to `spin_temperature` itself is exactly the same, but that is where it's failing in one case and not the other...

I'll try to whip up a patch for this today (I'll try have it get pulled into the doc-update branch, because otherwise it will be a nightmare).

**Comment by steven-murray** [comment]
Fixed in #70.
